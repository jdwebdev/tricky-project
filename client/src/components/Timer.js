import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { updateTimer } from '../actions/statistiques';
import PropTypes from 'prop-types';

let interval;

const Timer = (props) => {

    const count = useSelector(state => state.stats.timer);
    const dispatch = useDispatch();

    useEffect(() => {
        if (props.stopTimer) {
            clearInterval(interval);
        } else {
            interval = setInterval(() => {
                dispatch(updateTimer());
            }, 1000)
        }
    }, [props.stopTimer]);
    
  return (
    <div>Timer : {count}</div>
  )
}

Timer.propTypes = {
    stopTimer: PropTypes.bool.isRequired
}

export default React.memo(Timer);