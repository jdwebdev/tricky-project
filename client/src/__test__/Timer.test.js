/* eslint-disable no-undef */
import React from 'react'
import Timer from '../components/Timer'
// import { shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import {render} from '@testing-library/react'

describe("Timer component", () => {
  const initialState = { 
    stats: {
      timer: 0
    }
  };
  const mockStore = configureStore();
  let store = mockStore(initialState);

  it("renders Timer component without crashing", () => {
    render(
      <Provider store={store}>
        <Timer stopTimer={true}/>
      </Provider> 
    );
  })

  it("renders correctly", () => {
    const tree = render(
      <Provider store={store}>
        <Timer stopTimer={true} />
      </Provider> 
    );
    expect(shallowToJson(tree)).toMatchSnapshot()
  })
})