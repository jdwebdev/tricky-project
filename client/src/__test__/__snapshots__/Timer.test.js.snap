// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer component renders correctly 1`] = `null`;
